### PR TITLE
tests: vitest

### PR DESCRIPTION
**Pull request by kellyjosephprice**
| [![PR App][icn]][demo] | Fix RM-XYZ |
| :--------------------: | :--------: |

## 🧰 Changes

Ports our test suite to `vitest`!

This mostly just adds the configuration and skips any tests that aren't passing already. I also deleted any magic block tests! 😃

image

It does not convert the browser tests to `vitest`. I'd like to get them all passing before we do that???

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

File: __tests__/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -53,6 +106,247 @@ exports[`emojis 1`] = `
...
+exports[`export multiple Markdown renderers > renders HTML 1`] = `undefined`;
+
+exports[`export multiple Markdown renderers > renders MD 1`] = `
+"# Hello World
+"
+`;
+
+exports[`export multiple Markdown renderers > renders custom React components 1`] = `[Function]`;
+
+exports[`export multiple Markdown renderers > renders hAST 1`] = `undefined`;
```
Comment (kellyjosephprice) [inline __tests__/__snapshots__/index.test.js.snap:120]: Something is wrong with these tests. Not sure where the snapshots were????

**Comment by erunion** [comment]
you should also update this `extends` bit in the test `.eslintrc` file to use `@readme/eslint-config/testing/vitest` too

https://github.com/readmeio/markdown/blob/f289c1b3f90d75cb637e13361627b23144232cb2/__tests__/.eslintrc#L2

**Review by trishaprile — APPROVED** [review]
😮

**Review by jennspencer — APPROVED** [review]
noice!

**Pull request merged**